### PR TITLE
[action] spm - add `--enable-code-coverage` option to `spm` action `test` command

### PR DESCRIPTION
**Pull request by Ast3r10n**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Previously, the --enable-code-coverage option was only available for the generate-xcodeproj command.

Allowing its use for the test command would enable code coverage reports for the test command as well.

### Description

The --enable-code-coverage option is now also usable when using the test command.

### Testing Steps

Use the spm action with the test command and set the enable_code_coverage option to true: the command should now gather coverage reports.

### CODE REVIEW

File: fastlane/lib/fastlane/actions/spm.rb [hunk 1]
```diff
@@ -47,7 +47,7 @@ def self.available_options
                                        end),
           FastlaneCore::ConfigItem.new(key: :enable_code_coverage,
                                        env_name: "FL_SPM_ENABLE_CODE_COVERAGE",
-                                       description: "Enables code coverage for the generated Xcode project when using the generate-xcodeproj command",
+                                       description: "Enables code coverage for the test command and the generated Xcode project when using the generate-xcodeproj command",
```
Comment (rogerluan) [inline fastlane/lib/fastlane/actions/spm.rb:50]:
```suggestion
                                       description: "Enables code coverage for the generated Xcode project when using the 'generate-xcodeproj' and the 'test' command",
```

**Review by rogerluan — CHANGES_REQUESTED** [review]
Thanks for your contribution! 🙌 

Left a minor copy change request. Also could you also update the description on line 318 of `spm_spec.rb` to add "for generate-xcodeproj"? 😊  https://github.com/fastlane/fastlane/pull/17805/files#diff-05fa981c4fb9fd6ca1fe98ee8a29caa1350f6062592e64c81b4798b7d3ed6b69R318

**Review by rogerluan — APPROVED** [review]
Nice work! 🚀 code LGTM!

**Comment by Ast3r10n** [comment]
Hello @rogerluan, is this PR blocked by the unsuccessful checks? 
They actually failed for the new Homebrew requirements, so they should pass now. Would it be possible to re-run the checks to allow this PR to be merged?

**Comment by rogerluan** [comment]
@Ast3r10n actually I don't have access to our CI to retrigger it 😬 But yeah CI would need to pass. Can you push something to retrigger to see if it passes? Maybe merge master into this branch?

We might get this merged later this week 🙏

**Comment by Ast3r10n** [comment]
@rogerluan done, let's see what happens!

**Comment by rogerluan** [comment]
> @rogerluan done, let's see what happens!

It worked! 💪 thanks!

**Review by joshdholtz — APPROVED** [review]
This is 🔥 Thank you for the contribution! Really appreciate it ❤️

**Pull request merged**